### DISCUSSION TITLE
How to add Field Names for a record within a record - Avro Schema

### DISCUSSION THREAD

**Discussion by sujin-sam** [sngular/kloadgen#411] [Q&A]
I have a below sample schema, can you pls let me know how to add FieldName in Value Schema Serializer Config ?

`{"type": "record", "name": "AnEvent", "namespace": "com.sujin.event", "fields": [{ "name": "data", "type": {"type": "record", "name": "Data", "fields": [{"name": "appId", "type": {"type": "string", "avro.java.string": "String"}}]}}]}`

**Comment by sujin-sam** [comment]
Is it something like `data.Data.appId` or `data.appId` ?

**Reply by jemacineiras** [reply]
Our component has not an editor, so you cannot add new field into a schema. Only configure values to generate. If you need more fields, you need to modify the schema outside JMeter + Kloadgen, upload it to the Schema Registry and reload.
Btw, you already has `data.appId` field ... You cannot see it in the component?

**Reply by sujin-sam** [reply]
I tried loading schema file, it gave an error, so thought we need to add it manually

**Reply by sujin-sam** [reply]
We use jks files for schema registry SSL, but getting a bad cert error, it's working fine when I create a producer as Java program.
Do I need to add a schema registry property if I use jks ?

**Reply by jemacineiras** [reply]
So if you configure the Schema Registry component to access your Schema Registry server you don't get connected right? Propably we need to add those security configuration properties. If you have the cert files where you are running the JMeter
